### DISCUSSION TITLE
How to enable aws xray traces?

### DISCUSSION THREAD

**Discussion by mani9223-oss** [open-telemetry/opentelemetry-dotnet-instrumentation#3590] [Q&A] [answered]
By default, OpenTelemetry Collector has enabled exporting to AWS X-Ray without any additional configurations. However, I am not seeing any traces in the X-ray or errors in the logs. Are there any extra steps we need to do to get the traces in Xray?

**Comment by nrcventura** [comment]
Is this a question about exporting data from an OpenTelemetry collector? Or is this a question about getting data from a .net application sent to an OpenTelemetry collector?

**Comment by mani9223-oss** [comment]
Its about getting data from .net application sent to OpenTelemetry collector.

**Comment by nrcventura** [comment, marked as answer]
At this point the auto instrumentation project does not include out of the box support for many of the AWS libraries and their instrumentation libraries that can be found in the contrib repo, because of the dependencies that those libraries also introduce to this project. https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3401 has more details about what happened when that work was attempted previously. @birojnayak may have a little more information available about what can be done in the short term to help you out. For example, you may need to add the necessary dependencies to your application and manually configure things using the [plugin model](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/main/docs/plugins.md) for this project.

**Reply by birojnayak** [reply]
@nrcventura thanks ! @mani9223-oss what @nrcventura told is right.. there is no out of the box support for x-ray sampler in the autoinstrumentation repo (which I will see if I can bring next year or so), but our internal team has build a plugin [here](https://github.com/aws-observability/aws-otel-dotnet-instrumentation) out of the autoinstrumentation , which allows for x-ray sampler and soon we will be supporting it. I will ping my internal team to respond when this will be supported.

**Comment by mani9223-oss** [comment]
Thanks for the response @nrcventura @birojnayak
